dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java

```diff
@@ -0,0 +1,193 @@
+package com.dotcms.rest.api.v1.page;
+
+import com.dotmarketing.beans.ContainerStructure;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.VersionableAPI;
+import com.dotmarketing.business.web.HostWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.containers.business.ContainerAPI;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.languagesmanager.business.LanguageAPI;
+import com.dotmarketing.portlets.templates.business.TemplateAPI;
+import com.dotmarketing.portlets.templates.design.bean.TemplateLayout;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.VelocityUtil;
+import com.dotmarketing.viewtools.DotTemplateTool;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.liferay.portal.model.User;
+import org.apache.velocity.context.Context;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.Serializable;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provides the utility methods that interact with HTML Pages in dotCMS. These methods are used by
+ * the Page REST end-point.
+ *
+ * @author Will Ezell
+ * @author Jose Castro
+ * @version 4.2
+ * @since Oct 6, 2017
+ */
+public class PageResourceHelper implements Serializable {
+
+    private static final long serialVersionUID = 296763857542258211L;
+
+    private final HostWebAPI hostWebAPI = WebAPILocator.getHostWebAPI();
+    private final HTMLPageAssetAPI htmlPageAssetAPI = APILocator.getHTMLPageAssetAPI();
+    private final LanguageAPI languageAPI = APILocator.getLanguageAPI();
+    private final VersionableAPI versionableAPI = APILocator.getVersionableAPI();
+    private final TemplateAPI templateAPI = APILocator.getTemplateAPI();
+    private final ContainerAPI containerAPI = APILocator.getContainerAPI();
+
+    private static final boolean RESPECT_FE_ROLES = Boolean.TRUE;
+    private static final boolean RESPECT_ANON_PERMISSIONS = Boolean.TRUE;
+
+    /**
+     * Private constructor
+     */
+    private PageResourceHelper() {
+
+    }
+
+    /**
+     * Provides a singleton instance of the {@link PageResourceHelper}
+     */
+    private static class SingletonHolder {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)